### ISSUE TITLE
Raspberry Pi 4

### ISSUE THREAD

**Issue by mdPlusPlus** [swetoast/steamlink-launcher#13]
In the README.md it says:

>its intended for RPI 2 and 3 not 4

Could we get a quick explanation on why the Raspberry Pi 4 is not supported and whether it will be supported in the future?  

What work needs to be done to make it work on the Pi 4 (if possible at all)?

**Comment by swetoast** [comment]
it probly will be but for now its not

**Comment by swetoast** [comment]
its due to HW issues in the firmware

**Comment by mdPlusPlus** [comment]
When you say firmware you mean the one located in the Raspberry Pi's EEPROM?  
Are we talking about explicit bugs in the current firmware or have certain APIs/syscalls/whatever changed due to Video Core VI having a new hardware decoder?

**Comment by swetoast** [comment]
last time i check the checked it couldnt handle full hd without lagging but if you have a rpi4 feel free to try it out and report back

**Comment by mdPlusPlus** [comment]
Okay, this is still very vague. So the issue is with OSMC (and LibreELEC) not being able to utiliize hardware accelerated video decoding? I'm just trying to find out what I have to look out for.

Guess I'll just have to try it out.

**Comment by mdPlusPlus** [comment]
Until the OSMC test images are up, I'm trying to get this to work on LibreELEC: https://github.com/mdPlusPlus/steamlink-launcher

I was able to fix your installation script so it would install on `LibreELEC-RPi4.arm-9.80-Milhouse-20200602023748-#0601-g44747ec`.  

However, running `/storage/steamlink/steamlink.sh` results in the following error:  
`* failed to add service - already in use?`

Official responses in the Steam forums indicate you need X11 to run Steam Link.  
LibreELEC does not ship with X11 anymore, though.

Do you have any idea how to work around this? Would this require custom LibreELEC builds with X11 enabled? Or am I missing something and X11 isn't required at all under LibreELEC?

**Comment by swetoast** [comment]
Did u manage to get it running ?

**Comment by mdPlusPlus** [comment]
Not on a Raspberry Pi 4 running LibreELEC. However, I started a complete rewrite which currently installs and starts fine under Kodi Matrix on LibreELEC and OSMC on a Raspberry Pi 3: https://github.com/mdPlusPlus/steamlink-launcher/tree/rewrite/  
(I will probably put that branch into its own repository though.)

Next on the map is the installation under Kodi 19 on Raspberry Pi OS (formerly Raspbian). Then I will check whether the actual streaming works how it's supposed to.

**Comment by swetoast** [comment]
So feel free to push updates to my dev branch if you find issues :)

**Comment by mdPlusPlus** [comment]
I stumbled upon several issues which I tried to address in my `dev` and later the `systemd` branch: https://github.com/mdPlusPlus/steamlink-launcher/commits/systemd
I'm not sure which state that branch is currently in, since I focussed on the rewrite in Python.

I *think* it is in a working state on a RPi3 running LibreELEC Matrix and the only issue is autostart at boot, which can easily be fixed by adding `ConditionPathExists=/tmp/steamlink.ready` to the systemd service file and then actually writing `/tmp/steamlink.ready` before starting it.

However, I don't think I will put any more work into these branches, as I found the Python rewrite to be more flexible.
